packages/models/src/models/LivechatContacts.ts

```diff
@@ -126,6 +127,26 @@ export class LivechatContactsRaw extends BaseRaw<ILivechatContact> implements IL
 		return this.updateOne({ _id: contactId }, update, options);
 	}
 
+	async updateContactCustomFields(
+		contactId: string,
+		dataToUpdate: { customFields: Record<string, unknown>; conflictingFields: ILivechatContactConflictingField[] },
+		options?: FindOneAndUpdateOptions,
+	): Promise<ILivechatContact> {
+		if (!dataToUpdate.customFields && !dataToUpdate.conflictingFields) {
+			throw new Error('At least one of customFields or conflictingFields must be provided');
+		}
+
+		const updatedResult = await this.findOneAndUpdate(
+			{ _id: contactId },
+			{
+				$set: { ...dataToUpdate },
+			},
+			{ returnDocument: 'after', ...options },
+		);
+
+		return updatedResult as ILivechatContact;
```

Comment:
pls remove the type cast and fix the return type. 

typecasts should be avoided as a general rule.. in this case for example, what happens if `contactId` is not found? it **will** return a `null`, but since you typed the function as `Promise<ILivechatContact>` and used the type casting to adhere to the return type you're creating a potential bug in your code.

Reply:
> Small note: I followed the file's current pattern, though.

if you've found other places doing the same, pls go ahead and create a tech debt tasks so we can get rid of them.. we should always be looking into improving the code and not replicating bad code.